Sources/Markdown/Walker/Walkers/MarkupFormatter.swift

```diff
@@ -347,6 +347,11 @@ public struct MarkupFormatter: MarkupWalker {
 
         /// The current line number.
         var lineNumber = 0
+
+        /// The "effective" line number, taking into account any queued newlines which have not been printed.
```

Comment:
Just for my understanding – what value does the `lineNumber` property provide? It seems confusing to have both of these now. I wouldn't be sure which to use in a given scenario. Maybe we can expand the docs of `lineNumber` to explain where its useful? It seems like its doc comment, "The current line number", is maybe not accurate?

(I'm wondering if we should rename it to something like `lastLineNumberWithContent` but that's a little verbose and I imagine would cause a fair bit of churn in the repo...)

Reply:
Looking at the implementation, the `lineNumber` property is meant to represent the last line of *printed content*, and line breaks are queued up to be printed the next time non-blank content is printed. (I'm not sure the precise reason for this design decision, but it may have something to do with ensuring that line prefixes and leading indentation are correctly processed? This code is relatively old so i'd have to dig to find the original author.)

I'll update the doc comment to reflect the way `lineNumber` and `effectiveLineNumber` are used.